src/reducers/articleReducer.js

```diff
@@ -73,6 +79,72 @@ export default (state = initialState, action) => {
       return { ...state, message: action.payload };
     case GET_ALL_ARTICLES_FAILURE_MSG:
       return { ...state, message: action.payload };
+    case BOOKMARK:
+      return {
+        ...state,
+        response: {
+          ...state.response,
+          getOneArticle: {
+            ...state.response.getOneArticle,
+            bookmarked: true,
+          },
+        },
+      };
+
+    case BOOKMARK_ON_ALL:
+      return {
+        ...state,
+        all: {
+          ...state.all,
+          data: state.all.data.map((article) => {
+            if (article.id === payload.id) {
+              const modifiedArticle = { ...article };
+              modifiedArticle.bookmarked = true;
+              return modifiedArticle;
+            }
+            return article;
+          }),
+        },
+      };
+
+    case BOOKMARK_ERROR:
+      return {
+        ...state,
+        bookMarkmessage: action.payload,
+      };
+    case UNBOOKMARK:
+      return {
+        ...state,
+        response: {
+          ...state.response,
+          getOneArticle: {
+            ...state.response.getOneArticle,
+            bookmarked: false,
+          },
+        },
+      };
+
+    case UNBOOKMARK_ON_ALL:
+      return {
+        ...state,
+        all: {
+          ...state.all,
+          data: state.all.data.map((article) => {
+            if (article.id === payload.articleId) {
```

Comment:
Nice work here @oluwajuwon. But I think you can use destructuring here. What do you think?

Reply:
nice observation @okoroemeka  👍🏿. Will look into that